### ISSUE TITLE
Control enhancement: variable display should offer a scrollable table representation for compound values

### ISSUE THREAD

**Issue by codemanyak** [fesch/Structorizer.Desktop#439]
With the grown capabilities to work with large arrays (#14) and records (#423) in Structorizer, the display of such values in the Executor Control is often poor (resulting in an enormous single-line initializer expression). It would be helpful to have an opportunity to represent (and edit) values of this kind in a scrollable table (at least for the outermost level in a first approach).
I might look like in these preview screenshots:
![executorcontrol3 26-08](https://user-images.githubusercontent.com/15326471/31635703-ff675a4e-b2c7-11e7-9a76-0c461d11bb86.png)
After pressing the pulldown button of e. g. variable `reflexions` a table editor would pop up:
![executor_valueeditor_3 26-08a](https://user-images.githubusercontent.com/15326471/31635795-68874854-b2c8-11e7-95fc-26e09db1775f.png)
Unless the inspected complex object is a constant, the values in the right column can be edited. While the cell editor is active, the commit button is disabled (because the cell editor hasn't stored the change yet, also see #438):
![executor_valueeditor_3 26-08b](https://user-images.githubusercontent.com/15326471/31635899-cc439190-b2c8-11e7-9920-5c1e6ed9cb17.png)
Changes via the plain value line in the main variable display should still remain possible as alternative. The effect of changes via one of the ways is of course to be present on consecutive modifying activities via either way.  

On this occasion, the following changes to the behaviour of the Executor Control are proposed:

1. The width of the left column of the variable display table should be optimized in a way that it takes no more room than required for the longest variable name or the column header. The entire remaining width should be dedicated to the value column. (The same should apply to the popped up value editor tables.)
1. During the session, the last position of the control window should not always be forgotten. Instead, it should reopen at that very last position, which is usually by intent placed aside of the Structorizer main window, not in front of it.

**Comment by codemanyak** [comment]
Implemented as described. Example of the value editor for a record constant (not editable, differing icon, differing button bar).
![executor_valueeditor_3 26-08c](https://user-images.githubusercontent.com/15326471/31640597-a768f8da-b2df-11e7-8308-20c4e4a38f9d.png)

**Comment by GitMensch** [comment]
Looks nice. Wouldn't it be "relative easy" be possible to just store the current dialog values (not in the final object) and reopen it with pointing to a double clicked object (for example in your "reflexions" screenshot on the value of index 1 to open index 0/1 with were/was)?

**Comment by codemanyak** [comment]
@GitMensch I'm lost with your question. What exactly are you asking me? If you (double-)click on the right column of whatever row you may edit the value (unless it's part of a constant).
The changes you make in such a sub-editor are temporary until you either commit or discard them (all). Ony if you commit, the changed element or component values will be registered as value change on the main Control and adopted by Executor on resuming execution (with play or step button).
I have no clue what you mean wis "were/was".

**Comment by GitMensch** [comment]
> I have no clue what you mean wis "were/was".

@codemanyak I've meant the second row (index 1). If you double-klick on it the same window could be reopened showng this "table" with the two separate entries "index0 were" and "index1 was". If you leave this by either "discard" or "commit" you would return to the previous (possibly updated) window (object). This design for recursive lists looks easy (you also could open up another window on top with a style preventing the "parent" window to be changed, but this may be (or not) simpler).

**Comment by codemanyak** [comment]
@GitMensch Oh, now I see what you meant, of course. I thought of recursively applying the same mechanism i.e. opening another modal instance of the window on top the previous one for nested arrays/structs. Its modality would ensure that you cannot simultaneously change the same item on both levels. This is what I meant with "For a future release the concept might be extended to be recursive" - I simply didn't want to oblige myself to finish this just now.
(I plan to come to an end with the most important changes for release 3.27 now. This will include the conversion of PERFORMed paragraphs to subroutines in #354, but may postpone the COBOL generator.)

**Comment by GitMensch** [comment]
Yes, please postpone the COBOL generator (just deactivating it).

**Comment by codemanyak** [comment]
The recursive application of the pulldown buttons for the value editors should now be the next improvement.
Though of minor importance, the colour status of the buttons is to checked.

**Comment by codemanyak** [comment]
The recursive ValuePresenter is ready and waiting on branch [codemanyak/Structorizer.Desktop/master](https://github.com/codemanyak/Structorizer.Desktop/commit/2ef2a93ed5d216e5ee1b312a2c728fe5698f9345) for integration into the next version (3.27-01).
